olive/passes/onnx/conversion.py

```diff
@@ -412,26 +422,64 @@ def _export_ranked_model(params):
             restore_args = replace_tensor_parallel_layers()
 
             input_model = DistributedPyTorchModelHandler(**model_config)
-            olive_pytorch_model = input_model.load_model(local_rank)
-            dummy_inputs = olive_pytorch_model.get_dummy_inputs()
-            io_config = None if pass_config["use_dynamo_exporter"] else olive_pytorch_model.get_io_config()
-            pytorch_model = olive_pytorch_model.prepare_session(rank=local_rank)
 
-            with torch.no_grad():
-                ranked_onnx_model = OnnxConversion._export_pytorch_model(
-                    pytorch_model,
-                    dummy_inputs,
-                    io_config,
-                    pass_config,
-                    device,
-                    torch_dtype,
-                    tempdir,
-                )
+            if input_model.hf_config and input_model.hf_config.components:
+                ranked_models = []
+                component_names = []
+                for component_name, component_model in input_model.get_hf_components(local_rank):
+                    dummy_inputs = component_model.get_dummy_inputs()
+                    io_config = None if pass_config["use_dynamo_exporter"] else component_model.get_io_config()
+                    pytorch_model = component_model.prepare_session(rank=local_rank)
+
+                    with torch.no_grad():
+                        ranked_component_modelproto = OnnxConversion._export_pytorch_model(
+                            pytorch_model,
+                            dummy_inputs,
+                            io_config,
+                            pass_config,
+                            device,
+                            torch_dtype,
+                            tempdir,
+                        )
+
+                    ranked_component_model = model_proto_to_olive_model(
```

Comment:
if it's saved to `output_filepath`, wouldn't the components overwrite each other?

Reply:
Since, the components are going to be merge anyways, might be easier to just keep the component model protos in memory and call merge_decoder on them directly? That way, there is no need to save, load and clean up.

Reply:
No output_file is ranked model name of the format model_XX.onnx so each file is named differently i.e. based on the rank.

Reply:
I mean the components of the same rank. this code is using `output_filepath` for both components. So wouldn't the second component overwrite the second component?

Reply:
Wouldn't that be assuming that there are only two components, and they will be merged. There aren't use case yet but theoretically there could be more than two components and the result might be a _CompositeModel_. Of course, that isn't supported in this implementation but that's a potential scenario.

Reply:
we can ignore the keep modelproto suggestion. But there is still the issue of the components being saved to the same filepath.

Reply:
Fixed.

Reply:
Thanks! for the more than two component case, it will now be on our future selves to sort out the filepaths haha.